edi_storage_oca/models/edi_backend.py

```diff
@@ -0,0 +1,169 @@
+                "%s ignored: no storage and/or input directory specified.", self.name
+            )
+            return False
+
+        exchange_types = self.env["edi.exchange.type"].search(
+            self._storage_exchange_type_pending_input_domain()
+        )
+        for exchange_type in exchange_types:
+            # NOTE: this call might keep hanging the cron
+            # if the remote storage is slow (eg: too many files)
+            # We should probably run this code in a separate job per exchange type.
+            file_names = self._storage_get_input_filenames(exchange_type)
+            _logger.info(
+                "Processing exchange type '%s': found %s files to process",
+                exchange_type.display_name,
+                len(file_names),
+            )
+            for file_name in file_names:
+                self.with_delay()._storage_create_record_if_missing(
+                    exchange_type, file_name
+                )
+        return True
+
+    def _storage_exchange_type_pending_input_domain(self):
+        """Domain for retrieving input exchange types."""
+        return [
+            ("backend_type_id", "=", self.backend_type_id.id),
+            ("direction", "=", "input"),
+            "|",
+            ("backend_id", "=", False),
+            ("backend_id", "=", self.id),
+        ]
+
+    def _storage_create_record_if_missing(self, exchange_type, remote_file_name):
+        """Create a new exchange record for given type and file name if missing."""
+        file_name = os.path.basename(remote_file_name)
+        extra_domain = [("exchange_filename", "=", file_name)]
+        existing = self._find_existing_exchange_records(
+            exchange_type, extra_domain=extra_domain, count_only=True
+        )
+        if existing:
+            return
+        record = self.create_record(
+            exchange_type.code, self._storage_new_exchange_record_vals(file_name)
+        )
+        _logger.debug("%s: new exchange record generated.", self.name)
+        return record.identifier
+
+    def _storage_get_input_filenames(self, exchange_type):
+        full_input_dir_pending = exchange_type._storage_fullpath(
+            self.input_dir_pending
+        ).as_posix()
+        if not exchange_type.exchange_filename_pattern:
+            # If there is not pattern, return everything
+            # TODO: clean this up, .list_files is deprecated in fs_storage
+            filenames = [
+                x
+                for x in self.storage_id.list_files(full_input_dir_pending)
+                if x.strip("/")
+            ]
+            return filenames
+
+        bits = [exchange_type.exchange_filename_pattern]
+        if exchange_type.exchange_file_ext:
+            bits.append(r"\." + exchange_type.exchange_file_ext)
+        pattern = "".join(bits)
+        # TODO: clean this up, .find_files is deprecated in fs_storage
+        full_paths = self.storage_id.find_files(pattern, full_input_dir_pending)
```

Comment:
Hi @QuocDuong1306 and @simahawk 

I am testing this PR and the method `find_files` (https://github.com/OCA/storage/blob/1a4d80faaab0ddf6dfc80d867472dd6ce4777c5f/fs_storage/models/fs_storage.py#L373) is not returning the full path from the root for me. Maybe you know what I may be missing.

I have an FTP where the home folder contains /import folder to import files. When the call to the method is done like `self.fs.ls(relative_path, detail=False)` being relative_path "/import", I just get the filename of the file inside the folder.

Reply:
I'll have a look soon

Reply:
@JordiMForgeFlow are you using `directory_path`? If yes, you might have inconsistent results. See https://github.com/OCA/storage/issues/311

Reply:
@simahawk I am not using directory_path. Nevertheless, I have been testing and I have been able to get it working fine by assuming that find_files returns only the filename, not the full path.

The problem is mainly that, if for example I have a file _test_file.xml_ in a folder called _import_, the find_files method returns _test_file.xml_. Then, when the line `return [p[pending_path_len:].strip("/") for p in full_paths]` is executed, the filename stored in the exchange record is left as _ile.xml_ (stripping the first characters as it assumes that it is actually the full path)